vaadin-split-layout-flow-parent/vaadin-split-layout-flow/src/main/java/com/vaadin/flow/component/splitlayout/SplitLayout.java

```diff
@@ -294,17 +282,30 @@ public void setSecondaryStyle(String styleName, String value) {
         setInnerComponentStyle(styleName, value, false);
     }
 
+    /**
+     * Returns the component if the given components array contains only one or
+     * a wrapper div with the given components if the array contains more.
+     *
+     * @param components
+     *            the components to wrap
+     * @return the component or a wrapper div
+     */
+    private Component getComponentOrWrap(Component... components) {
+        if (components.length == 1) {
+            return components[0];
+        } else {
+            Div container = new Div();
+            container.add(components);
+            return container;
+        }
+    }
+
     private void setComponent(Component component, String slot) {
         Component child = component == null ? new Div() : component;
+        SlotUtils.clearSlot(this, slot);
         SlotUtils.addToSlot(this, slot, child);
```

Comment:
nit: could be shortened to `SlotUtils.setSlot(this, slot, child);`

Reply:
Thanks! Done.